extensions/amp-date-picker/1.0/component/component.tsx

```diff
@@ -0,0 +1,92 @@
+import * as Preact from '#preact';
+import {useMemo} from '#preact';
+import {forwardRef} from '#preact/compat';
+import {FunctionalComponent, Ref} from '#preact/types';
+
+import {DateRangePicker} from './date-range-picker';
+import {SingleDatePicker} from './single-date-picker';
+import {DatePickerContext} from './use-date-picker-context';
+
+import {
+  DEFAULT_END_INPUT_SELECTOR,
+  DEFAULT_INPUT_SELECTOR,
+  DEFAULT_LOCALE,
+  DEFAULT_MONTH_FORMAT,
+  DEFAULT_ON_ERROR,
+  DEFAULT_START_INPUT_SELECTOR,
+  DEFAULT_WEEK_DAY_FORMAT,
+  ISO_8601,
+} from '../constants';
+import {getCurrentDate} from '../date-helpers';
+import {BentoDatePickerProps} from '../types';
+
+type PropertiesWithDefaults =
+  | 'endInputSelector'
+  | 'format'
+  | 'inputSelector'
+  | 'locale'
+  | 'maximumNights'
+  | 'minimumNights'
+  | 'mode'
+  | 'monthFormat'
+  | 'numberOfMonths'
+  | 'onError'
+  | 'startInputSelector'
+  | 'today'
+  | 'type'
+  | 'weekDayFormat';
+
+const DEFAULT_PROPS: Pick<BentoDatePickerProps, PropertiesWithDefaults> = {
```

Comment:
I wanted to have a way to ensure type safety for all these props in the base components, but provide defaults in the parent. There might be easier ways to do this (such as maintaining separate type definitions with optional and required props), but this solution using Pick and Omit also worked pretty well.